### DISCUSSION TITLE
Hotspot Problem

### DISCUSSION THREAD

**Discussion by emincanyuksel** [google/model-viewer#4963] [Q&A]
Hello, I am adding a hotspot via the model viewer, but it only gives me the data-surface code and not the data-position. What should I do? Can you help me?

**Comment by elalish** [comment]
a hotspot needs *either* `data-position` *or* `data-surface`. Is the surface not working? If you need position instead, you can use [`positionAndNormalFromPoint()`](https://modelviewer.dev/docs/index.html#entrydocs-annotations-methods-positionAndNormalFromPoint).

**Reply by emincanyuksel** [reply]
Okay, I'm looking for a code for the text on the hotspot seen in the picture. It should not grow or shrink when zoomed in, it should remain at a fixed size, but I couldn't do it, please help.

![IMG-20241202-WA0009](https://github.com/user-attachments/assets/003fe9ff-7209-4afa-8b54-6c81ed7f2ac4)

**Reply by elalish** [reply]
That looks like a screenshot from our editor. Did you look at the HTML snippet it generates on the upper-right? You'll need to scroll to see it all. Alternatively, have you clicked "Download Scene" and inspected the complete HTML page that's downloaded?

**Reply by emincanyuksel** [reply]
The model in this picture is uploaded on my site, I leave the link. If you can't see the link, you can watch the video. ([Link](https://kentkart.emincanyuksel.com.tr/kentkart_fuar_alani/index.html))

https://github.com/user-attachments/assets/ae3db280-dd85-497c-ba20-d6a5273cb725

**Reply by elalish** [reply]
It looks like you succeeded - the text does not grow or shrink when you zoom in and out. What problem are you hoping to fix?

**Reply by emincanyuksel** [reply]
It is obvious in the video that I want it to be a fixed size and be the same size from close up and from a distance.

**Reply by elalish** [reply]
I think you mean you want the text to be sized relative to the 3D scene instead of a fixed 2D text size? You can try to set the CSS font size based on the camera radius from the camera-change event, but it won't be perfect. The text is 2D HTML, not part of the 3D scene. In general that approach is dangerous for readability and a11y since the text can easily get too small to read.

**Reply by emincanyuksel** [reply]
Well, is there any sample work or code that I can use to do this?
